app/src/main/java/com/dessalines/thumbkey/utils/Types.kt

```diff
@@ -15,6 +15,7 @@ data class KeyboardDefinitionModes(
 
 data class KeyboardDefinitionSettings(
     val autoCapitalizers: AutoCapitalizers = arrayOf(),
+    val layoutForceNoAutoShift: Boolean = false,
```

Comment:
`autoShift: Boolean = true` would be better.

Reply:
This one still

Reply:
used `layoutAllowAutoShift`.